_posts/2019-04-24-istio-service-mesh-s2s.md

```diff
@@ -0,0 +1,431 @@
+
+## Show me the code
+
+Explaining the Istio service to service authentication and authorization will now be explain by using an example setup.
+Note that the code snippets have been shortened in this blogpost.
+This is denoted with three dots `...`.
+The full examples can be found in the accompanying repository on [Github](https://github.com/pietervincken/istio-service-to-service-demo){:target="_blank" rel="noopener noreferrer"}
+
+### Prerequisites
+
+This demo assumes that Istio is already installed in the cluster with the demo profile enabled. 
+See [Install Istio](https://istio.io/docs/setup/kubernetes/install/helm/#option-1-install-with-helm-via-helm-template){:target="_blank" rel="noopener noreferrer"} for more information on the installation of Istio.
+In the demo repository, a small [script](https://github.com/pietervincken/istio-service-to-service-demo/blob/master/1-install-istio.sh){:target="_blank" rel="noopener noreferrer"} can be found that can assist in setting up the demo environment
+
+### Setup
+
+The setup of our application is a very simple service with a database backend. 
+Our service exposes one HTTP GET endpoint which will be accessed by the outside world. 
+Our database is an Apache CouchDB instance.
+Both the database and the service run inside Kubernetes.
+The setup is shown in the image below.
+
+<img class="image fit" style="margin:0px auto; max-width: 600px;" src="{{ '/img/2019-04-14-istio-service-mesh-s2s/setup.png' | prepend: site.baseurl }}" alt="demo setup" />
+
+### Create namespace
+
+First, a new namespace is created.
+The service and database will both be added to this namespace.
+
+```bash
+kubectl create namespace with-istio
+kubectl label namespace with-istio istio-injection=enabled
+```
+
+### Install CouchDB
+
+Next, the database is installed.
+This looks like a normal stateful set for a CouchDB database.
+There are some important changes.
+
+First, a specific service account is created for CouchDB. 
+This is needed as Istio will use the service accounts in Kubernetes as its identities.
+The service account is linked to the podspec in the stateful set definition. 
+This way, it can be used by the Istio proxy later on.
+
+Secondly, the probes have been adapted to work in Istio.
+Since Istio intercepts all traffic in the pod, it will also intercept requests from the Kube API to the service.
+Since the demo setup requires mTLS to be used, the probes would fail because the Kube API doesn't use mTLS.
+Instead of using changing the probes, Istio now has the option to rewrite the probes during the automatic proxy injection. 
+More information on the probes can be found in the [Istio Docs](https://istio.io/help/ops/setup/app-health-check/#probe-rewrite){:target="_blank" rel="noopener noreferrer"}.
+
+Note that no Istio specific configuration is required in the service manifests.
+This is possible because the demo profile automatically enables the sidecar injection for all namespaces.
```

Comment:
This is possible because the demo profile automatically enables the sidecar *injector* **and we enabled the injection on the `with-istio` namespace using the `istio-injection=enabled` label.**

Reply:
Updated